test/plugin_helper/test_server.rb

```diff
@@ -796,6 +797,51 @@ class Dummy < Fluent::Plugin::TestBase
     end
   end
 
+  sub_test_case "UDPServer" do
```

Comment:
Since this `sub_test_case` focuses on `max_bytes`, it should be reflected in the `sub_test_case` name.
Otherwise it might be hard to recognize the difference with `sub_test_case '#server_create_udp'`.

Reply:
Sorry for confusing you, but please let me explain this since perhaps you don't get the intent of this code.

Actually, this `sub_test_case` is for `EventHandler.UDPServer`.
I wanted to create a test that focused on it.
(I ended up using `server_create()` for initialization, though,,,)

https://github.com/fluent/fluentd/blob/206b46b91560177366371c1f05bbcf09ac006d86/lib/fluent/plugin_helper/server.rb#L518

So the test cases are `With sock` and `Without sock`.

> Since this `sub_test_case` focuses on `max_bytes`, it should be reflected in the `sub_test_case` name.

So this test is not focused on `max_bytes`, actually.
The data `Normal` is for the tests simply for `UDPServer.on_readable_without_sock` and `UDPServer.on_readable_with_sock`.
(`max_bytes` is specified just because it is a mandatory option.)

My intention is something like this.
But, certainly, it may be confusing.

Now, what do you think?

Would changing the name of the subtest like this improve this problem?

```
sub_test_case "UDPEventHandler::Server"
```

Reply:
The design of `Fluent::PluginHelper::Server` doesn't seem intend to expose `EventHandler.UDPServer`, it just only intend to use via `server_create` or `server_create_udp`. In actual you don't use  `EventHandler.UDPServer` directly, you use it via `server_create` in this test. In my opinion users don't need to aware the implementation detail `EventHandler.UDPServer`, so that test don't need to be aware it too.

It seems that these tests should be placed under `sub_test_case '#server_create_udp'`.

Reply:
Oh, I see. Thanks.

It means that there is not much benefit in trying to make unit tests of `EventHandler` right?

I will try to fix it that way.

Reply:
> It means that there is not much benefit in trying to make unit tests of `EventHandler` right?

That's right.

Reply:
Thanks for your advice!
Fixed!